init/mkdumprd

```diff
@@ -87,6 +68,48 @@ function run_dracut()
     eval "bash -$- $DRACUT $DRACUT_ARGS"
 }                                                                          # }}}
 
+#
+# Build $INITRD for fadump with dump capture support using dracut            {{{
+function build_fadumprd()
+{
+    readonly FADUMP_TMPDIR="$(mktemp -d -t fadump.XXXXXX)"
+    readonly FADUMPRD_TMPDIR="$FADUMP_TMPDIR/fadumproot"
+    readonly TARGET_INITRD="$INITRD"
+
+    INITRD="$FADUMP_TMPDIR/fadump.img"
+    run_dracut
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        exit $ret
+    fi
+
+    mkdir $FADUMPRD_TMPDIR
+    if ! (pushd "$FADUMPRD_TMPDIR" > /dev/null && lsinitrd --unpack $INITRD && popd > /dev/null); then
+        echo "Failed to unpack dump capture scripts."
+        exit 1
+    fi
+
+    DRACUT_ARGS="--force --add zz-fadumpinit --omit kdump"
+
+    REBUILD_PARAMS=$(lsinitrd $TARGET_INITRD /lib/dracut/build-parameter.txt)
```

Comment:
What happens if $TARGET_INITRD does not exist at this point?

Reply:
REBUILD_PARAMS would be empty as lsinitrd does not put anything on stdout.. 
While there would be an error printed on stderr, that can simple be ignored and $TARGET_INITRD is built with "--force --add zz-fadumpinit --omit kdump" dracut parameters..